### ISSUE TITLE
Inconsistent description of rd field in mov instruction

### ISSUE THREAD

**Issue by gallottino** [RIP-Comm/clementine#24]
In arm7tdmi.rs line 97:

```
// bits [24-21] are the RD
let rd = (op_code & 0x00_00_F0_00) >> 12;
```

The comment is inconsistent: RD is in [15-12] bits.
Reference [here](https://problemkaputt.de/gbatek.htm#armopcodesdataprocessingalu).

**Comment by FedericoBruzzone** [comment]
I have written some detailed documentation about this issue. 
Let me know if you want me to do the PR 👍

**Comment by gallottino** [comment]
Yes, you can proceed with PR.

**Comment by gallottino** [comment]
Solved here #29
